mod_ci/controllers.py

```diff
@@ -369,33 +369,29 @@ def kvm_processor(db, kvm_name, platform, repository, delay):
         test_branch.checkout(True)
 
         # TODO: check what happens on merge conflicts
-        # Rebase on master
-        # try:
-        #     repo.git.rebase('master')
-        # except GitCommandError:
-        #     progress = TestProgress(
-        #         test.id, TestStatus.preparation,
-        #         'Rebase on master'
-        #     )
-        #     db.add(progress)
-        #     progress = TestProgress(
-        #         test.id, TestStatus.canceled,
-        #         'Merge conflict, please resolve.'
-        #     )
-        #     db.add(progress)
-        #     db.commit()
-        #     # Report back
-        #     gh_commit = repository.statuses(test.commit)
-        #
-        #     with app.app_context():
-        #         target_url = url_for(
-        #             'test.by_id', test_id=test.id, _external=True)
-        #     context = "CI - %s" % test.platform.value
-        #     gh_commit.post(
-        #         state=Status.ERROR, description='Failed to rebase',
-        #         context=context, target_url=target_url)
-        #     # Return, so next one can be handled
-        #     return
+        # Merge on master if no conflict
+        try:
+            repo.git.merge('test_branch', '--ff-only')
+        except GitCommandError:
+            tests = Test.query.filter(Test.pr_nr == test.pr_nr).all()
```

Comment:
This is not necessary. We're in a method to process a specific test, so it only needs cancellation for that specific test. Given that this method is only called if the test isn't started yet, a check for the progress isn't necessary either :)